packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/editInfoUtils/applyChange.ts

```diff
@@ -0,0 +1,81 @@
+import checkEditInfoState, { ImageEditInfoState } from './checkEditInfoState';
+import deleteEditInfo from './deleteEditInfo';
+import generateDataURL from './generateDataURL';
+import getEditInfoFromImage from './getEditInfoFromImage';
+import getGeneratedImageSize from './getGeneratedImageSize';
+import ImageEditInfo from '../types/ImageEditInfo';
+import saveEditInfo from './saveEditInfo';
+import { IEditor, PluginEventType } from 'roosterjs-editor-types';
+
+/**
+ * @internal
+ * Apply changes from the edit info of an image, write result to the image
+ * @param editor The editor object that contains the image
+ * @param image The image to apply the change
+ * @param editInfo Edit info that contains the changed information of the image
+ * @param previousSrc Last src value of the image before the change was made
+ * @returns True if the image is changed, otherwise false
+ */
+export default function applyChange(
+    editor: IEditor,
+    image: HTMLImageElement,
+    editInfo: ImageEditInfo,
+    previousSrc: string
+): boolean {
+    let newSrc = '';
+
+    const initEditInfo = getEditInfoFromImage(image);
+    const state = checkEditInfoState(editInfo, initEditInfo);
+
+    switch (state) {
+        case ImageEditInfoState.ResizeOnly:
+            // For resize only case, no need to generate a new image, just reuse the original one
+            newSrc = editInfo.src;
+            break;
+        case ImageEditInfoState.SameWithLast:
+            // For SameWithLast case, image may be resized but the content is still the same with last one,
+            // so no need to create a new image, but just reuse last one
+            newSrc = previousSrc;
+            break;
+        case ImageEditInfoState.FullyChanged:
```

Comment:
What happens when the case is "Invalid"?

Reply:
It means we got dirty data which may not be generated from our code. So we need to ignore it